examples/Leveraging_model_distillation_to_fine-tune_a_model.ipynb

```diff
@@ -0,0 +1,1055 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "81ddec71-c2a6-411a-aecd-6556cddc36c8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def process_example(index, row, model, df):\n",
+    "    global progress_index\n",
+    "    \n",
+    "    try:\n",
+    "        # Generate the prompt using the row\n",
+    "        prompt = generate_prompt(row, varieties)\n",
+    "\n",
+    "        df.at[index, model + \"-variety\"] = call_model(model, prompt)\n",
+    "        \n",
+    "        # Print progress only for every 20th row\n",
+    "        if progress_index % 20 == 0:\n",
+    "            print(f\"Question processed {progress_index}/{len(df)}\")\n",
+    "        \n",
+    "        progress_index += 1\n",
+    "    except Exception as e:\n",
+    "        print(f\"Error processing model {model}: {str(e)}\")\n",
+    "\n",
+    "\n",
+    "def process_dataframe(df, model):\n",
+    "    global progress_index\n",
+    "    progress_index = 1  # Reset progress index\n",
+    "\n",
+    "    # Process each example concurrently using ThreadPoolExecutor\n",
+    "    with concurrent.futures.ThreadPoolExecutor() as executor:\n",
+    "        futures = {executor.submit(process_example, index, row, model, df): index for index, row in df.iterrows()}\n",
+    "        \n",
+    "        for future in concurrent.futures.as_completed(futures):\n",
+    "            try:\n",
+    "                future.result()  # Wait for each example to be processed\n",
+    "            except Exception as e:\n",
+    "                print(f\"Error processing example: {str(e)}\")\n",
+    "\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "680587d1-bcb8-45bd-bfe8-1f771ae1a5bd",
+   "metadata": {},
+   "source": [
+    "Let's try out our call model function before processing the whole dataframe and check the output."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "95915fd2-24f6-4908-a54f-9ce59073233d",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'Bordeaux-style Red Blend'"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "answer = call_model('gpt-4o', generate_prompt(df_france_subset.iloc[0], varieties))\n",
+    "answer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "941b9027-eb22-4eed-84de-8b962fe4fee2",
+   "metadata": {},
+   "source": [
+    "Great! We confirmed we can get a grape variety as an output, let's now process the dataset with both `gpt-4o` and `gpt-4o-mini` and compare the results. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "1cedf978-e60b-4f56-9c8c-da6b0a722320",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
```

Comment:
I would suggest using tqdm for this instead of print statements as it looks a bit prettier.